app/scripts/models/reliers/oauth.js

```diff
@@ -16,9 +16,49 @@ define(function (require, exports, module) {
   var RelierKeys = require('lib/relier-keys');
   var Url = require('lib/url');
   var Validate = require('lib/validate');
+  var Vat = require('lib/vat');
 
   var RELIER_FIELDS_IN_RESUME_TOKEN = ['state', 'verificationRedirect'];
 
+  /*eslint-disable camelcase*/
+  var CLIENT_INFO_SCHEMA = {
+    id: Vat.hex().required().as('clientId'),
```

Comment:
This is more a comment about your `vat` library, but it is a little bit ambiguous to me as a reader which way round the transformation flows. Is it from `id => clientId` or `clientId => id`? Is there anything you can do to make that clearer?